test/models/bet.test.js

```diff
@@ -0,0 +1,315 @@
+      delete input.resultIndex;
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if resultIndex is null', () => {
+      input.resultIndex = null;
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if resultIndex is undefined', () => {
+      input.resultIndex = undefined;
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if resultIndex is string', () => {
+      input.resultIndex = "12";
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if resultIndex is array', () => {
+      input.resultIndex = [2];
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if resultIndex is object', () => {
+      input.resultIndex = {id:2};
+      assert.throws(() => new Bet(input), Error, "resultIndex must be a Number");
+    });
+
+    it('It should throw if eventRound is missing', () => {
+      delete input.eventRound;
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+
+    it('It should throw if eventRound is null', () => {
+      input.eventRound = null;
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+
+    it('It should throw if eventRound is undefined', () => {
+      input.eventRound = undefined;
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+
+    it('It should throw if eventRound is string', () => {
+      input.eventRound = "12";
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+
+    it('It should throw if eventRound is array', () => {
+      input.eventRound = [2];
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+
+    it('It should throw if eventRound is object', () => {
+      input.eventRound = {id:2};
+      assert.throws(() => new Bet(input), Error, "eventRound must be a Number");
+    });
+  });
+
+  describe('format', () => {
+    let input;
+
+    beforeEach(() => {
+      input = {};
+      Object.assign(input, rawInput);
+    });
+
+    it('it should format all the fields', () => {
+      const bet = new Bet(input);
+      assert.equal(bet.txid, input.txid);
+      assert.equal(bet.txStatus, input.txStatus);
+      assert.equal(bet.blockNum, input.blockNum);
+      assert.equal(bet.eventAddress, input.eventAddress);
+      assert.equal(bet.betterAddress, input.betterAddress);
+      assert.equal(bet.resultIndex, input.resultIndex);
+      assert.equal(bet.amount, input.amount);
+      assert.equal(bet.eventRound, input.eventRound);
+      assert.equal(bet.txType, TX_TYPE.BET);
+    });
+
+    it('It should format the type to be vote', () => {
```

Comment:
You don't need to do all the other checks. Tests should be focused on what it's testing. You're testing the `txType` here so just test that and leave the other asserts out. The test before handles the other cases.

Reply:
please handle for all other tests